### PR TITLE
doc: mention exception for JoinGuard.detach() behavior

### PR DESCRIPTION
**Pull request by tshepang**

### CODE REVIEW

**Comment by aturon** [comment]
Note, this API has since been revamped.

**Comment by tshepang** [comment]
Yeah, this PR is obsolete, and the text is much nicer:

>  The child thread may outlive the parent (unless the parent thread is the main thread; the whole process is terminated when the main thread finishes.)
